### DISCUSSION TITLE
Expect changing machine's id not to reset graph's layout in VS Code

### DISCUSSION THREAD

**Discussion by nmtigor** [statelyai/xstate#4047] [Ideas]
In VS Code, when I change a machine's id, the graph's layout in XState Editor seems to be reset. I have to restore the layout manually, which is tedious for large graph. I hope the layout could be kept after changing the machine's id. Thanks!

**Comment by chaitanyaj2222** [comment]
This truly seems like an issue @nmtigor. [This](https://github.com/statelyai/xstate-tools) is the URL for xstate tooling github repo. I think that it is the right place for issues that are related to Xstate VScode extension.

**Comment by nmtigor** [comment]
See [xstate-tools/issues/347](https://github.com/statelyai/xstate-tools/issues/347)

**Reply by chaitanyaj2222** [reply]
Cool!
